common/src/stack/report-system/command/report/system/tests/test_partitions.py

```diff
@@ -0,0 +1,77 @@
+import pytest
+import math
+from collections import namedtuple, defaultdict
+from stack import api
+from stack.commands.report.system.fixtures.disk_partition_check import verify_disk
+
+@pytest.mark.parametrize('hostname',
+list(set([host['host'] for host in api.Call('list host partition')]))
+)
+class TestStoragePartition:
+
+# Test if the way stacki configured the disks is still how they are partitioned
+
+	def test_storage_partition(self, hostname, verify_disk):
+
+		# Get stacki storage config
+		storage_config = api.Call(f'list host storage partition {hostname}')
+		check_config = defaultdict(dict)
+		partition_info = namedtuple('partition', ['name', 'mountpoint', 'label', 'fstype', 'size'])
+		errors = []
+
+		if not storage_config:
+			pytest.skip(f'Could not get stacki partition config for host {hostname}')
+
+		# Convert stacki storage data structure to one
+		# pytest fixture can use
+		for partition in storage_config:
+			disk = partition['device']
+			disk_num = str(partition['partid'])
+			part_label = ''
+
+			if partition['mountpoint'] is None:
+				partition['mountpoint'] = ''
+
+			if partition['fstype'] is None:
+				partition['fstype'] = ''
+
+			try:
+				for option in partition['options'].split(' '):
+					if '--label=' in option:
+						part_label = option.split('--label=')[1]
+
+			except (KeyError, IndexError):
+				pass
+
+			check_config[partition['device']][disk + disk_num] = partition_info(
+					name = disk + disk_num, mountpoint = partition['mountpoint'],
+					label = part_label, fstype = partition['fstype'], size = partition['size']
+			)
+
+
+		# Call fixture function to find partitions that don't match the config
+		match_partitions = verify_disk(check_config, hostname, check_all_disks=True)
+
+		# Output errors
+		if match_partitions:
```

Comment:
This `if` statement is not needed as iterating over an empty dictionary with a for loop will result in iterating zero times.